lib/src/signed_video_h26x_auth.c

```diff
@@ -115,6 +115,112 @@ decode_sei_data(signed_video_t *self, const uint8_t *payload, size_t payload_siz
   return status;
 }
 
+/**
+ * Computes the GOP hash from |nalu_hash_list| and compares it with the GOP hash received from the
+ * signature.
+ *
+ * @param self A pointer to the signed_video_t structure.
+ * @return true if the computed GOP hash matches the received GOP hash and the status is SV_OK;
+ * false otherwise.
+ */
+static bool
+verify_gop_hash(signed_video_t *self)
+{
+  gop_info_t *gop_info = self->gop_info;
+  const size_t hash_size = self->verify_data->hash_size;
+  svrc_t status = SV_UNKNOWN_FAILURE;
+
+  // Compute the GOP hash from |nalu_hash_list|.
+  status = compute_partial_gop_hash(
+      self, gop_info->nalu_hash_list, gop_info->nalu_list_idx, gop_info->computed_gop_hash);
+  // Compare the computed GOP hash with the received GOP hash.
+  bool hashes_match =
+      (memcmp(gop_info->computed_gop_hash, self->received_gop_hash, hash_size) == 0);
+
+  // Verify the GOP hash if both the hashes match and the status is SV_OK.
+  gop_info->successfully_verified_gop_hash = (hashes_match && (status == SV_OK));
+
+  return gop_info->successfully_verified_gop_hash;
```

Comment:
Returning a set member just adds redundancy. Either use a `void` as return value or  update the member outside, or maybe you don't even need a member.

Reply:
I have make the function boolean because I am using calling the function